### PR TITLE
cross platform: add deb package creation support

### PR DESCRIPTION
**Pull request by obastemur**

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -116,6 +118,11 @@ while [[ $# -gt 0 ]]; do
         MAKE=0
         ;;
 
+	--create-deb=*)
```
Comment (dilijev) [inline build.sh]: nit : indentation

File: build.sh [hunk 2]
```diff
@@ -228,6 +235,34 @@ fi
...
+        cp $DEB_FOLDER/../ch $DEB_FOLDER/usr/local/bin/
+        if [[ $STATIC_LIBRARY == "" ]]; then
+            cp $DEB_FOLDER/../*.so $DEB_FOLDER/usr/local/bin/
+        fi
+        echo -e "Package: ChakraCore"\
+            "\nVersion: ${CREATE_DEB}"\
+            "\nSection: base"\
+            "\nPriority: optional"\
+            "\nArchitecture: amd64"\
+            "\nDepends: libc6 (>= 2.19)"\
```
Comment (digitalinfinity) [inline build.sh]: While we get the ICU story straightened out- don't we need libicu here? Our build instructions also have libuuid/libunwind/liblttng-ust (maybe for the PAL?) should we add them here?

File: build.sh [hunk 3]
```diff
@@ -228,6 +235,34 @@ fi
...
+            "\nMaintainer: ChakraCore <chakracore@microsoft.com>"\
+            "\nDescription: Microsoft ChakraCore"\
```
Comment (dilijev) [inline build.sh]: Should probably include the ®: "Microsoft ® ChakraCore" or "Chakra Core" to be consistent with the metadata in the Windows binary.

File: build.sh (hunk 2 above)
Comment (obastemur) [inline build.sh]: no icu needed. As for the others, we should find a solution similar to icu though. That is just for the warning, it doesn't have much effect as dpkg has no control over repos.

File: build.sh (hunk 3 above)
Comment (obastemur) [inline build.sh]: isn't 'Microsoft ChakraCore' informative enough for description?

Comment (dilijev) [inline build.sh]: I think the registered trademark symbol (®) with Microsoft might be a legal requirement or recommendation for any user-visible string which includes the company and product name.

As for ChakraCore versus Chakra Core I don't personally have a preference. Was just pointing out the inconsistency.

**Comment by obastemur** [comment]
@dilijev @digitalinfinity thanks for the reviews.

File: build.sh [hunk 4]
```diff
@@ -228,6 +235,34 @@ fi
...
+        fi
+        echo -e "Package: ChakraCore"\
+            "\nVersion: ${CREATE_DEB}"\
+            "\nSection: base"\
+            "\nPriority: optional"\
+            "\nArchitecture: amd64"\
+            "\nDepends: libc6 (>= 2.19), uuid-dev (>> 0), libunwind-dev (>> 0), libicu-dev (>> 0)"\
+            "\nMaintainer: ChakraCore <chakracore@microsoft.com>"\
+            "\nDescription: Chakra Core"\
+            "\n Open source Core of Chakra Javascript Engine"\
```
Comment (dilijev) [inline build.sh]: @liminzhu @EdMaurer @bterlson Does this metadata look alright for a public release?

**Comment by obastemur** [comment]
Merging this. We are not releasing this package yet. Necessary text additions can be made later.

**Pull request merged**